File: bench/stress/s5_fused.go
// S5 — fused polynomial pipeline, Go. go run s5_fused.go
package main

import "fmt"

func poly(x int64) int64 {
	return ((x%97)*31 + (x%13)*7 + 3) % 1000
}

func main() {
	var n int64 = 50_000_000
	var total int64
	for x := int64(0); x < n; x++ {
		p := poly(x)
		if p%2 == 0 {
			total += p
		}
	}
	fmt.Println(total)
}

File: bench/stress/s5_fused.rs
// S5 — fused polynomial pipeline, Rust. rustc -O s5_fused.rs -o s5_fused
fn poly(x: i64) -> i64 {
    ((x % 97) * 31 + (x % 13) * 7 + 3) % 1000
}

fn main() {
    let n: i64 = 50_000_000;
    let mut total: i64 = 0;
    for x in 0..n {
        let p = poly(x);
        if p % 2 == 0 {
            total += p;
        }
    }
    println!("{}", total);
}
